perf-tool/src/agentOptions.cpp

```diff
@@ -0,0 +1,281 @@
+            check_jvmti_error(jvmti, error, "Unable to enable VM ObjectAlloc event notifications.");
+        }
+        else if (!command.compare("stop"))
+        { // c == stop
+            printf("Obect Alloc Capability already disabled");
+            error = jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_VM_OBJECT_ALLOC, (jthread)NULL);
+            check_jvmti_error(jvmti, error, "Unable to disable ObjectAlloc event.");
+        }
+        else
+        {
+            invalidCommand(function, command);
+        }
+    }
+}
+
+void modifyMonitorStackTrace(std::string function, std::string command)
+{
+    // enable stack trace
+    if (!command.compare("start"))
+    {
+        setMonitorStackTrace(true);
+    }
+    else if (!command.compare("stop"))
+    {
+        setMonitorStackTrace(false);
+    }
+    else
+    {
+        invalidCommand(function, command);
+    }
+}
+
+//object alloc capability already enabled
+void modifyMethodEntryEvents(std::string function, std::string command, int sampleRate)
+{
+    jvmtiError error;
+    setMethodEntrySampleRate(sampleRate);
+    if (!command.compare("stop"))
+    {
+        error = jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_METHOD_ENTRY, (jthread)NULL);
+        check_jvmti_error(jvmti, error, "Unable to disable MethodEntry event.\n");
+    }
+    else if (!command.compare("start"))
+    { // c == start
+        error = jvmti->SetEventNotificationMode(JVMTI_ENABLE, JVMTI_EVENT_METHOD_ENTRY, (jthread)NULL);
+        check_jvmti_error(jvmti, error, "Unable to enable MethodEntry event notifications.\n");
+    }
+    else
+    {
+        invalidCommand(function, command);
+    }
+}
+
+void modifyExceptionBackTrace(std::string function, std::string command)
+{
+    // enable stack trace
+    if (!command.compare("start"))
+    {
+        setExceptionBackTrace(true);
+    }
+    else if (!command.compare("stop"))
+    {
+        setExceptionBackTrace(false);
+    }
+    else
+    {
+        invalidCommand(function, command);
+    }
+}
+
+void modifyExceptionEvents(std::string function, std::string command, int sampleRate)
+{
+    jvmtiError error;
+    setExceptionSampleRate(sampleRate);
+
+    if (!command.compare("start"))
+    {
+        error = jvmti->SetEventNotificationMode(JVMTI_ENABLE, JVMTI_EVENT_EXCEPTION, (jthread)NULL);
+        check_jvmti_error(jvmti, error, "Unable to enable Exception event notifications.\n");
+    }
+    else if (!command.compare("stop"))
+    {
+        error = jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_EXCEPTION, (jthread)NULL);
+        check_jvmti_error(jvmti, error, "Unable to disable Exception event.\n");
+    }
+    else
+    {
+        invalidCommand(function, command);
+    }
+}
+
+
+void agentCommand(json jCommand)
+{
+    jvmtiCapabilities capa;
+    jvmtiError error;
+    jvmtiPhase phase;
+
+    std::string function;
```

Comment:
You can combine these two lines into one:
`std::string function = jCommand["functionality"].get<std::string>();`
Similarly in other places.
Not sure, but `get` may actually return a reference to a const object.